searx/network/client.py

```diff
@@ -155,56 +192,376 @@ def get_transport(verify, http2, local_address, proxy_url, limit, retries):
+                    # > For situations when context block usage is not practical,
+                    # > it is possible to enter "manual mode" by sending a Request
+                    # > instance using client.send(..., stream=True).
+                    request = self.client.build_request(
+                        method=method,
+                        url=url,
+                        content=kwargs.get("content"),
+                        data=kwargs.get("data"),
+                        files=kwargs.get("files"),
+                        json=kwargs.get("json"),
+                        params=kwargs.get("params"),
+                        headers=kwargs.get("headers"),
+                        cookies=kwargs.get("cookies"),
+                        timeout=timeout,
+                        extensions=self.extensions,
+                    )
+                    response = self.client.send(
+                        request,
+                        stream=True,
+                        follow_redirects=kwargs.get("follow_redirects", False),
+                        auth=kwargs.get("auth"),
+                    )
+                else:
+                    response = self.client.request(method, url, extensions=self.extensions, timeout=timeout, **kwargs)
+                self._patch_response(response)
+                return response
+            except httpx.RemoteProtocolError as e:
+                if retry >= 0:
+                    # the server has closed the connection:
+                    # try again without decreasing the retries variable & with a new HTTP client
+                    self._reconnect_client()
+                    if self.logger:
+                        self.logger.warning('httpx.RemoteProtocolError: the server has disconnected, retrying')
+                    continue
+                raise e
+            except (httpx.RequestError, httpx.HTTPStatusError) as e:
+                raise e
+        return response  # type: ignore
+
+    def close(self):
+        self.client.close()
+
+    @property
+    def is_closed(self) -> bool:
+        return self.client.is_closed
+
+    def _new_client(self):
+        limit = httpx.Limits(
+            max_connections=self.max_connections,
+            max_keepalive_connections=self.max_keepalive_connections,
+            keepalive_expiry=self.keepalive_expiry,
+        )
+        # See https://www.python-httpx.org/advanced/#routing
+        mounts = {}
+        for pattern, proxy_url in self.proxies.items():
+            if not self.enable_http and pattern.startswith('http://'):
+                continue
+            if (
+                proxy_url.startswith('socks4://')
+                or proxy_url.startswith('socks5://')
+                or proxy_url.startswith('socks5h://')
+            ):
+                mounts[pattern] = _get_transport_for_socks_proxy(
+                    self.verify, self.enable_http2, self.local_address, proxy_url, limit, 0
```

Comment:
Comment: `0` is the number of retries on purpose set to this value. The `NetworkContext` classes manage the retries (and may use different HTTPClient)